### PR TITLE
Rental #create_reservation and #delete_reservation tests

### PR DESCRIPTION
**Pull request by cmednieks**
Addresses coverage issues brought up in #108 and #105

### CODE REVIEW

File: spec/models/rental_spec.rb [hunk 1]
```diff
@@ -223,4 +223,31 @@
     rent = create :mock_rental, item_type: create(:item_type, name: 'Test 220', base_fee: 200, fee_per_day: 20)
     expect(FinancialTransaction.where(rental: rent).map(&:amount)).to eq([220])
   end
+
+  describe '#delete_reservation' do
+    context 'error thrown' do
+      it 'logs the error and returns false' do
+        r = build(:rental)
+        r.create_reservation
```
Comment (phantompatrol) [inline spec/models/rental_spec.rb]: Are the two lines needed, by any chance? Or are the callbacks not firing? What I am suggesting is something along the lines of...

```Ruby
it 'logs the error and returns false' do
  allow(Inventory).to receive(:delete_reservation).and_raise(InventoryExceptions::AuthError)
  expect(create(:rental).delete_reservation).to be false
end
```

File: spec/models/rental_spec.rb [hunk 2]
```diff
@@ -223,4 +223,31 @@
...
+      it 'logs error and returns false for a series of errors' do
+        r = build(:rental)
+        allow(Inventory).to receive(:create_reservation).and_raise(InventoryExceptions::InventoryError)
+        expect(r.create_reservation).to be false
+        r = build(:rental)
+        allow(Inventory).to receive(:create_reservation).and_raise(InventoryExceptions::ReservationError)
+        expect(r.create_reservation).to be false
+        r = build(:rental)
+        allow(Inventory).to receive(:create_reservation).and_raise(InventoryExceptions::AuthError)
+        expect(r.create_reservation).to be false
```
Comment (phantompatrol) [inline spec/models/rental_spec.rb]: Ditto to the above comment

File: spec/models/rental_spec.rb (hunk 1 above)
Comment (cmednieks) [inline spec/models/rental_spec.rb]: They do appear to be necessary; I tried to do the above code and it seems that building a rental does not trigger `create_reservation` . I've noticed while working with the unfriendly errors that the callbacks don't seem to be working the way that we've been expecting them to and am trying to fix that problem in an upcoming PR.

**Comment by tomecho** [comment]
@cmednieks please resolve merge conflict with master

**Pull request merged**